### PR TITLE
pe-parse: init at 1.2.0

### PR DESCRIPTION
**Pull request by arcz**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/pe-parse/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "trailofbits";
+    repo = "pe-parse";
+    rev = "v${version}";
+    sha256 = "1jvfjaiwddczjlx4xdhpbgwvvpycab7ix35lwp3wfy44hs6qpjqv";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
```
Comment (r-burns) [inline pkgs/development/libraries/pe-parse/default.nix:15]: Could be a candidate for multiple outputs if you wish to reduce dependers' closure size
```nix
  outputs = [ "out" "dev" "bin" ];

  postInstall = ''
    moveToOutput bin $bin
  '';
```

File: pkgs/development/libraries/pe-parse/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  doCheck = true;
+  checkPhase =
+    let var = if stdenv.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH";
+    in ''
+      export ${var}=./pe-parser-library
+      dump-pe/dump-pe ../test/assets/example.exe
+    '';
```
Comment (r-burns) [inline pkgs/development/libraries/pe-parse/default.nix:22]: Becomes a bit simpler as an installCheck:
```suggestion
  doInstallCheck = true;
  installCheckPhase = ''
    $bin/bin/dump-pe ../test/assets/example.exe
  '';
```

Comment (arcz) [reply pkgs/development/libraries/pe-parse/default.nix]: that's a great tip, thanks!

File: pkgs/development/libraries/pe-parse/default.nix (hunk 1 above)
Comment (arcz) [reply pkgs/development/libraries/pe-parse/default.nix:15]: I had some trouble building another library this way. Maybe I'll leave the single output for now as it's not big and refactor that later if needed.

**Review by r-burns — APPROVED** [review]
Result of `nixpkgs-review pr 107506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pe-parse</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 107506` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pe-parse</li>
  </ul>
</details>

**Review by thiagokokada — COMMENTED** [review]
Tested with a random Windows binary, seems to work.

However I think leaving this in `pkgs/development/libraries/` kinda strange. Wouldn't it make more sense to put this in `pkgs/development/tools/`? :thinking:

**Comment by arcz** [comment]
pe-parse is a library, the dump-pe program is rather an extra thing

**Review by thiagokokada — APPROVED** [review]

**Review by cole-h — CHANGES_REQUESTED** [review]

**Review by cole-h — APPROVED** [review]
Diff LGTM.

---

Result of `nixpkgs-review pr 107506` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pe-parse</li>
  </ul>
</details>

**Pull request merged**